subprojects/core/src/main/java/org/gradle/process/internal/worker/request/WorkerAction.java

```diff
@@ -83,7 +85,7 @@ public void execute(WorkerProcessContext workerProcessContext) {
                     // Make the argument serializers available so work implementations can register their own serializers
                     registration.add(RequestArgumentSerializers.class, argumentSerializers);
                     registration.add(InstantiatorFactory.class, instantiatorFactory);
-                    registration.add(InternalProblems.class, new DefaultProblems(ImmutableList.of(new WorkerProblemEmitter(responder)), CurrentBuildOperationRef.instance()));
+                    registration.add(InternalProblems.class, new DefaultProblems(ImmutableList.of(new WorkerProblemEmitter(responder)), null,  CurrentBuildOperationRef.instance(), new DefaultBuildTreeExceptionProblemContainer(new DefaultBuildSessionExceptionProblemContainer())));
```

Comment:
🤔I know Reinhold is going to beat me, but I would make the `ProblemContainer` nullable

Reply:
I don't like `@Nullable` either. At the same time I've significantly reduced the amount of times we invoke `DefaultProblems.<init>`.